x-pack/plugins/security/server/config.ts

```diff
@@ -24,6 +20,114 @@ const providerOptionsSchema = (providerType: string, optionsSchema: Type<any>) =
     schema.never()
   );
 
+type ProvidersCommonConfigType = Record<
+  'enabled' | 'showInSelector' | 'order' | 'description',
+  Type<any>
+>;
+function getCommonProviderSchemaProperties(overrides: Partial<ProvidersCommonConfigType> = {}) {
+  return {
+    enabled: schema.boolean({ defaultValue: true }),
+    showInSelector: schema.boolean({ defaultValue: true }),
+    order: schema.number({ min: 0 }),
+    description: schema.maybe(schema.string()),
+    ...overrides,
+  };
+}
+
+function getUniqueProviderSchema(
+  providerType: string,
+  overrides?: Partial<ProvidersCommonConfigType>
+) {
+  return schema.maybe(
+    schema.recordOf(schema.string(), schema.object(getCommonProviderSchemaProperties(overrides)), {
+      validate(config) {
+        if (Object.values(config).filter(provider => provider.enabled).length > 1) {
+          return `Only one "${providerType}" provider can be configured.`;
+        }
+      },
+    })
+  );
+}
+
+type ProvidersConfigType = TypeOf<typeof providersConfigSchema>;
+const providersConfigSchema = schema.object(
+  {
+    basic: getUniqueProviderSchema('basic', {
```

Comment:
__note:__ while working on this I realized we can now join `basic` and `token` into one provider and enable token based authentication with something like `xpack.security.authc.basic.basic1.useTokens: true`. I'll file an issue for that.